src/amo/reducers/versions.js

```diff
@@ -234,6 +246,90 @@ const reducer = (
       };
     }
 
+    case LOAD_ADDONS_BY_AUTHORS:
+    case LOAD_ADDON_RESULTS:
+    case LOAD_COLLECTION_ADDONS:
+    case LOAD_CURRENT_COLLECTION:
+    case LOAD_CURRENT_COLLECTION_PAGE:
+    case LOAD_RECOMMENDATIONS:
+    case SEARCH_LOADED: {
+      const { addons, results } = action.payload;
+
+      const items = addons || results;
+      const newVersions = {};
+      for (const addon of items) {
+        // For collection related actions, the addon is available in addon.addon.
+        const addonToUse = addon.addon || addon;
+        if (addonToUse.current_version) {
+          const version = createInternalVersion(addonToUse.current_version);
+          newVersions[version.id] = version;
+        }
+      }
+
+      return {
+        ...state,
+        byId: {
+          ...state.byId,
+          ...newVersions,
+        },
+      };
+    }
+
+    case LOAD_HOME_ADDONS: {
+      const { collections, shelves } = action.payload;
+
+      const newVersions = {};
+      for (const shelf of Object.keys(shelves)) {
+        for (const addon of shelves[shelf].results) {
+          if (addon.current_version) {
+            const version = createInternalVersion(addon.current_version);
+            newVersions[version.id] = version;
+          }
+        }
+      }
+
+      for (const collection of collections) {
+        if (collection && collection.results) {
+          for (const addon of collection.results) {
+            const version = createInternalVersion(addon.addon.current_version);
+            newVersions[version.id] = version;
+          }
+        }
+      }
+
+      return {
+        ...state,
+        byId: {
+          ...state.byId,
+          ...newVersions,
+        },
+      };
+    }
+
+    case LANDING_LOADED: {
+      const { featured, highlyRated, trending } = action.payload;
+
+      const newVersions = {};
+      for (const apiResponse of [featured, highlyRated, trending]) {
+        if (apiResponse) {
```

Comment:
It would be nice to get some test coverage for when `apiResponse` is falsy. I commented out this statement and I don't see any failures.

Reply:
[None of these can be falsey](https://github.com/mozilla/addons-frontend/blob/2018.10.18-1/src/amo/actions/landing.js#L21-L29), so I can just remove this check.